src/note/mod.rs

```diff
@@ -14,6 +10,9 @@ use time::OffsetDateTime;
 /// Alias for [`u64`]. Uniquely identifies a note.
 pub type NoteId = u64;
 
+/// Metadata is a HashMap of String to JSON values constructed from the front matter
+pub type Metadata = HashMap<String, serde_json::Value>;
```

Comment:
Making this public only clutters the API. You can make it internally public by changing to

```rust
pub(crate) type Metadata = HashMap<String, serde_json::Value>;
```

However, I say we drop the type aliasing and just use the explicit type everywhere. Add `use serde_json::Value;` and use `HashMap<String, Value>` instead of `Metadata`.

Reply:
Sounds good to me 😄 

I don't think this can become crate-private because it needs to be accessible from importing projects so that they can access the extra data. Unless we want to add a method like `Node::extra(&self, key: &str) -> Option<&Value>` and completely hide the field.